src/helpers/credential.rs

```diff
@@ -269,6 +269,59 @@ impl<'a> Credential<'a> {
         Ok(())
     }
 
+    /// Revokes a given credential with the help of vade and updates revocation list credential
+    ///
+    /// # Arguments
+    /// * `credential_str` - credential to be revoked in seralized string format
+    /// * `updated_key_jwk` - public key in jwk format to sign did update
+    /// * `private_key` - bbs private key to sign revocaton request
+    ///
+    /// # Returns
+    /// * `String` - the result of updated revocation list doc after credential revocation
+    #[cfg(feature = "plugin-did-sidetree")]
+    pub async fn revoke_credential(
+        &mut self,
+        credential_str: &str,
+        update_key_jwk: &str,
+        private_key: Option<&str>,
+    ) -> Result<String, CredentialError> {
+        let credential: BbsCredential = serde_json::from_str(credential_str)?;
+        let revocation_list: RevocationListCredential = self
+            .get_did_document(&credential.credential_status.revocation_list_credential)
+            .await?;
+
+        let proving_key = match private_key {
```

Comment:
Using `credential.issuer` fallback wouldn't work properly at the moment. It should be safer to make `private_key` mandatory instead of using the fallback here. ^^

Reply:
done , made it mandatory to pass private key